src/components/utils/test/thread_test.cc

```diff
@@ -175,6 +247,175 @@ TEST_F(ThreadTest, StartOneThreadTwice_ExpectTheSameThreadStartedTwice) {
   EXPECT_TRUE(delegate_->check_value());
 }
 
+TEST(ThreadTest2, StartAndJoinForce_Success) {
+  AutoLock test_lock(test_mutex_);
+
+  // Create a delegate with infinite threadMain
+  InfiniteMainTestDelegate* delegate = new InfiniteMainTestDelegate();
+  Thread* thread = CreateThread(kThreadName.c_str(), delegate);
+
+  // Default ThreadOptions means it is joinable
+  EXPECT_TRUE(thread->start(threads::ThreadOptions()));
+  EXPECT_TRUE(thread->is_joinable());
+
+  // Wait for the threadMain to signal that it has started
+  cond_var_.Wait(test_lock);
+  EXPECT_TRUE(delegate->check_value());
+
+  thread->join(Thread::JoinOptionStop::kForceStop);
+  EXPECT_FALSE(thread->is_running());
+
+  delete delegate;
+  DeleteThread(thread);
+}
+
+TEST(ThreadTest2, StartAndJoinNoForce_Success) {
+  AutoLock test_lock(test_mutex_);
+
+  // Create a delegate with infinite threadMain
+  InfiniteMainTestDelegate* delegate = new InfiniteMainTestDelegate();
+  Thread* thread = CreateThread(kThreadName.c_str(), delegate);
+  EXPECT_TRUE(thread->start(threads::ThreadOptions()));
+  EXPECT_TRUE(thread->is_joinable());
+
+  // Wait for the threadMain to signal that it has started
+  cond_var_.Wait(test_lock);
+
+  // Signal the delegate it has to finish the execution of threadMain
+  delegate->notified_ = true;
+  thread->join(Thread::JoinOptionStop::kNoStop);
+
+  EXPECT_FALSE(thread->is_running());
+  EXPECT_TRUE(delegate->check_value());
+
+  delete delegate;
+  DeleteThread(thread);
+}
+
+TEST(ThreadTest2, StartStopStart_Success) {
+  // Create a delegate with infinite threadMain
+  InfiniteMainTestDelegate* delegate = new InfiniteMainTestDelegate();
+  Thread* thread = CreateThread(kThreadName.c_str(), delegate);
+  EXPECT_TRUE(thread->start(threads::ThreadOptions()));
+  EXPECT_TRUE(thread->is_joinable());
+
+  // Wait for the thread to notify it has started execution
+  {
+    AutoLock test_lock(test_mutex_);
+    while (!delegate->notify_running_) {
+      cond_var_.Wait(test_lock);
+    }
+  }
+  EXPECT_TRUE(thread->is_running());
+
+  // Signal the thread to stop and wait for the execution
+  thread->stop();
+  thread->join(Thread::JoinOptionStop::kNoStop);
```

Comment:
@dev-gh I want to test the `stop()`. Calling `join()` is just to make sure the thread has finished.